controllers/storagecluster/storageconsumer.go

```diff
@@ -130,3 +108,132 @@ func (s *storageConsumer) ensureDeleted(_ *StorageClusterReconciler, _ *ocsv1.St
+	volumeSnapshotClassNames[util.GenerateNameForSnapshotClass(storageCluster.Name, util.CephfsSnapshotter)] = true
+
+	if storageCluster.Spec.NFS != nil && storageCluster.Spec.NFS.Enable {
+		volumeSnapshotClassNames[util.GenerateNameForCephNetworkFilesystemStorageClass(storageCluster)] = true
+	}
+
+	// for day2 volumesnapshotclasses
+	volumeSnapshotClassesInCluster := &snapapi.VolumeSnapshotClassList{}
+	if err := kubeClient.List(ctx, volumeSnapshotClassesInCluster); err != nil {
+		return nil, err
+	}
+	for idx := range volumeSnapshotClassesInCluster.Items {
+		// TODO: skip volumesnapshotclasses that are from external mode if both internal & external mode is enabled
+		vsc := &volumeSnapshotClassesInCluster.Items[idx]
+		vscName := vsc.Name
+		vscDriver := vsc.Driver
+		if slices.Contains(supportedCsiDrivers, vscDriver) && !volumeSnapshotClassNames[vscName] {
+			volumeSnapshotClassNames[vscName] = true
+		}
+	}
+
+	vscSpec := make([]ocsv1a1.VolumeSnapshotClassSpec, len(volumeSnapshotClassNames))
+	idx := 0
+	for vscName := range maps.Keys(volumeSnapshotClassNames) {
+		vscSpec[idx] = ocsv1a1.VolumeSnapshotClassSpec{Name: vscName}
+		idx++
+	}
+	return vscSpec, nil
+}
+
+func getVolumeGroupSnapshotClassesForDistribution(ctx context.Context, kubeClient client.Client, storageCluster *ocsv1.StorageCluster) (
+	[]ocsv1a1.VolumeGroupSnapshotClassSpec, error) {
+
+	volumeGroupSnapshotClassNames := map[string]bool{}
+	volumeGroupSnapshotClassNames[util.GenerateNameForGroupSnapshotClass(storageCluster, util.RbdGroupSnapshotter)] = true
+	volumeGroupSnapshotClassNames[util.GenerateNameForGroupSnapshotClass(storageCluster, util.CephfsGroupSnapshotter)] = true
+
+	crd := &metav1.PartialObjectMetadata{}
+	crd.SetGroupVersionKind(extv1.SchemeGroupVersion.WithKind("CustomResourceDefinition"))
+	crd.Name = VolumeGroupSnapshotClassCrdName
+	if err := kubeClient.Get(ctx, client.ObjectKeyFromObject(crd), crd); client.IgnoreNotFound(err) != nil {
+		return nil, err
+	}
+	if crd.UID != "" {
+		// for day2 volumegroupsnapshotclasses
+		volumeGroupSnapshotClassesInCluster := &groupsnapapi.VolumeGroupSnapshotClassList{}
+		if err := kubeClient.List(ctx, volumeGroupSnapshotClassesInCluster); err != nil {
+			return nil, err
+		}
+		for idx := range volumeGroupSnapshotClassesInCluster.Items {
+			// TODO: skip volumegroupsnapshotclasses that are from external mode if both internal & external mode is enabled
+			vgsc := &volumeGroupSnapshotClassesInCluster.Items[idx]
+			vgscName := vgsc.Name
+			vgscDriver := vgsc.Driver
+			if slices.Contains(supportedCsiDrivers, vgscDriver) && !volumeGroupSnapshotClassNames[vgscName] {
+				volumeGroupSnapshotClassNames[vgscName] = true
+			}
+		}
+	}
+
+	vgscSpec := make([]ocsv1a1.VolumeGroupSnapshotClassSpec, len(volumeGroupSnapshotClassNames))
+	idx := 0
+	for vscName := range maps.Keys(volumeGroupSnapshotClassNames) {
+		vgscSpec[idx] = ocsv1a1.VolumeGroupSnapshotClassSpec{Name: vscName}
+		idx++
+	}
```

Comment:
As the wanted capacity is known there is no point in using an index. Using append is more idiomatic and with a known capacity will not re-allocate
```suggestion
	vgscSpec := make([]ocsv1a1.VolumeGroupSnapshotClassSpec, 0, len(volumeGroupSnapshotClassNames))
	for vscName := range maps.Keys(volumeGroupSnapshotClassNames) {
		vgscSpec = append(vgscSpec, ocsv1a1.VolumeGroupSnapshotClassSpec{Name: vscName})
	}
```

Reply:
done.